gsa/src/web/pages/tasks/dialog.js

```diff
@@ -225,6 +181,57 @@ const TaskDialog = ({
   const scanner = get_scanner(scanners, scanner_id);
   const scanner_type = isDefined(scanner) ? scanner.scannerType : undefined;
 
+  const [configType, setConfigType] = useState('openvas');
+  const [prevConfigType, setPrevConfigType] = useState('openvas');
+
+  // eslint-disable-next-line no-shadow
+  const handleScannerChange = (value, name) => {
+    // eslint-disable-next-line no-shadow
+    const scanner = get_scanner(scanners, value);
+    // eslint-disable-next-line no-shadow
+    const scanner_type = isDefined(scanner) ? scanner.scannerType : undefined;
+
+    if (
+      scanner_type === OPENVAS_SCANNER_TYPE ||
+      scanner_type === OSP_SENSOR_SCANNER_TYPE ||
+      scanner_type === GMP_SCANNER_TYPE
+    ) {
+      setConfigType('openvas');
+    } else if (scanner_type === OSP_SCANNER_TYPE) {
+      setConfigType('osp');
+    } else {
+      setConfigType('other');
+    }
+
+    log.debug('on scanner change', value, scanner);
```

Comment:
Should be removed. I've added it because the whole scanner and config changes are very complicated as you also noticed while working on this PR ;-)
```suggestion
```